### PR TITLE
Oop hw advanced by Oleh Kravets

### PR DESCRIPTION
**Pull request by Ketch92**

### CODE REVIEW

File: src/main/java/core/basesyntax/figure/figures/Circle.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        return radius;
+    }
+
+    public void setRadius(int radius) {
+        this.radius = radius;
+    }
+
+    @Override
+    public String toString() {
+        return getColor() + " " + NAME
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figure/figures/Circle.java:41]:
```suggestion
        return getColor() + " " + getClass().getSimpleName()
```
there is no need to create enum

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import core.basesyntax.figure.figures.Circle;
+import core.basesyntax.figure.figures.IsoscelesTrapezoid;
+import core.basesyntax.figure.figures.Rectangle;
+import core.basesyntax.figure.figures.RightTriangle;
+import core.basesyntax.figure.figures.Square;
+import java.util.Random;
+
+public class FigureProducer {
+
+    public static Figure getRandomFigure() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figure/producers/FigureProducer.java:13]:
```suggestion
    public Figure getRandomFigure() {
```
let's not use static methods here

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            case 4:
+            default:
+                return new IsoscelesTrapezoid(ColorProducer.getRandomColor(),
+                        randomizer.nextInt(bound),
+                        randomizer.nextInt(bound),
+                        randomizer.nextInt(bound));
+        }
+    }
+
+    public static Figure[] getRandomFigures() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figure/producers/FigureProducer.java:41]: same with other methods

Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figure/producers/FigureProducer.java:41]: we have duplicated code here, you can just hardcode amount value and call your getRandomFigures(int amount) method from here

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        int index = 0;
+
+        while (index < figures.length) {
+            figures[index++] = getRandomFigure();
+        }
+
+        return figures;
+    }
+
+    public static void print(Figure figure) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figure/producers/FigureProducer.java:65]: it's not logical to have print method in FigureProducer class, it has different purpose according to its name

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/figure/figures/Circle.java (hunk 1 above)
Comment (Ketch92) [reply src/main/java/core/basesyntax/figure/figures/Circle.java:41]: I thought that enum isn't fitting there. Ok thanks

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java (hunk 2 above)
Comment (Ketch92) [reply src/main/java/core/basesyntax/figure/producers/FigureProducer.java]: we shall not use static methods there!

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java (hunk 3 above)
Comment (Ketch92) [reply src/main/java/core/basesyntax/figure/producers/FigureProducer.java]: Nice tip, thx

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java (hunk 4 above)
Comment (Ketch92) [reply src/main/java/core/basesyntax/figure/producers/FigureProducer.java]: I though who produces also is able to print, but as you say

File: src/main/java/core/basesyntax/figure/figures/Circle.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/core/basesyntax/figure/figures/Circle.java]: we don't need to use enum here getClass().getSimpleName() is just a method

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java (hunk 4 above)
Comment (RomanDubovskyi) [reply src/main/java/core/basesyntax/figure/producers/FigureProducer.java]: if you have producer class it should create new objcets/values, if you need to print something you can do in the class where you call print, or create some util class. This way you follow Single responsibility principle

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import core.basesyntax.figure.figures.Square;
+import java.util.Random;
+
+public class FigureProducer {
+
+    public Figure getRandomFigure() {
+        Random randomizer = new Random();
+        int bound = 1000;
+
+        switch (new Random().nextInt(5)) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/figure/producers/FigureProducer.java:17]: if 5 is a number of your figures it's better to declare as a constant 
private static int FIGURE_NUMBER = 5;
this way it won't be a magic number in the code

**Review by RomanDubovskyi — APPROVED** [review]
cool, left last comment

File: src/main/java/core/basesyntax/figure/producers/FigureProducer.java (hunk 5 above)
Comment (Ketch92) [reply src/main/java/core/basesyntax/figure/producers/FigureProducer.java:17]: Ok, got it. Also had something like that on mind, I mean to get read of magic numbers.
Is there a way to find a number of classes who extends that class so the figure number will update it self?

**Pull request closed without merging**